pro/src/ui-kit/formV2/TextInput/TextInput.spec.tsx

```diff
@@ -80,12 +97,104 @@ describe('TextInput', () => {
   })
 
   it('should ignore onKeyDown event for type number', async () => {
-    render(<TextInput type="number" label="Input" name="test1" />)
+    render(
+      <TextInput type="number" label="Input" name="test1" required={true} />
+    )
 
     const input = screen.getByLabelText('Input *')
     await userEvent.type(input, '1000')
     await userEvent.keyboard('ArrowUp')
 
     expect(input).toHaveValue(1000)
   })
+
+  it('renders the input field with the correct label', () => {
+    render(<TextInput {...defaultProps} />)
+
+    // Check if the input is rendered with the label and placeholder
+    const input = screen.getByLabelText(/email address/i) // Match case-insensitively
+    expect(input).toBeInTheDocument()
+    expect(input).toHaveAttribute('placeholder', 'Enter your email')
+  })
+
+  it('renders the input field with a mandatory asterisk when required', () => {
+    render(<TextInput {...defaultProps} />)
+
+    // Check if the label has the asterisk for required fields
+    const label = screen.getByText('Email Address *')
+    expect(label).toBeInTheDocument()
+  })
+
+  it('renders the error message when the error prop is provided', () => {
+    const errorMessage = 'This field is required'
+    render(<TextInput {...defaultProps} error={errorMessage} />)
+
+    // Check if the error message is displayed
+    const error = screen.getByRole('alert')
+    expect(error).toHaveTextContent(errorMessage)
+  })
+
+  it('does not display error when no error prop is passed', () => {
+    render(<TextInput {...defaultProps} />)
+
+    // Check if the error is not displayed
+    const error = screen.queryByRole('alert')
+    expect(error).toBeNull()
+  })
+
+  it('disables the input when the disabled prop is passed', () => {
+    render(<TextInput {...defaultProps} disabled />)
+
+    // Check if the input is disabled
+    const input = screen.getByLabelText(/email address/i)
+    expect(input).toBeDisabled()
+  })
+
+  it('handles readonly mode correctly', () => {
+    render(<TextInput {...defaultProps} readOnly value="readonly value" />)
+
+    // Check if the input field is readonly
+    const readOnly = screen.getByText(/email address/i)
+    expect(readOnly).toBeInTheDocument()
+  })
+
+  it('should handle decimal input correctly with hasDecimal=true', () => {
+    render(<TextInput {...defaultProps} type="number" hasDecimal={true} />)
+
+    const input = screen.getByLabelText(/email address/i)
+
+    // Test if decimal input is allowed
+    fireEvent.change(input, { target: { value: 123.45 } })
+    expect(input).toHaveValue(123.45)
+
+    // Test if non-numeric input is rejected
+    fireEvent.keyDown(input, { key: 'a' })
+    expect(input).toHaveValue(123.45) // Input remains unchanged for non-numeric input
```

Comment:
`fireEvent` est plutôt déconseillé. Utiliser `userEvent` à la place.

```suggestion
    const user = userEvent.setup();

    // Test if decimal input is allowed
    await user.type(input, '123.45')
    expect(input).toHaveValue(123.45)

    // Test if non-numeric input is rejected
    await user.keyboard('a')
    expect(input).toHaveValue(123.45) // Input remains unchanged for non-numeric input
```